### PR TITLE
Add v2 view recovery code page

### PR DESCRIPTION
**Pull request by IniZio**
ref #3571

dependencies:
- [x] #3652

Desktop|Mobile
-|-
Screenshot 2024-01-22 at 13-40-25 Authgear|Screen Shot 2024-01-22 at 13 40 31

### CODE REVIEW

File: authui/src/authflowv2/components/recovery-code-list.css [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    --recovery-code-list__color: var(--color-grey-700);
+    --recovery-code-list__line-height: 1rem;
+    --recovery-code-list__letter-spacing: 0.025rem;
+
+    --recovery-code-list__btn-bg-color: var(--color-grey-700);
+    --recovery-code-list__btn-border-radius: 6.25em;
+    --recovery-code-list__btn-px: 1.5em;
+    --recovery-code-list__btn-py: calc(
+      var(--recovery-code-list__btn-font-size) * 13 / 14
+    );
```
Comment (tung2744) [inline authui/src/authflowv2/components/recovery-code-list.css:18]: Can we use em? And also please document how you constructed this formula.

File: authui/src/authflowv2/components.css [hunk 2]
```diff
@@ -21,3 +21,4 @@
 @import "./components/password-policy.css";
 @import "./components/icon.css";
 @import "./components/checkbox.css";
+@import "./components/recovery-code-list.css";
```
Comment (tung2744) [inline authui/src/authflowv2/components.css:24]: There is another stylesheet `code-block.css`  which seems having the same purpose, try to reuse it or delete it if it is not needed?

File: resources/authgear/templates/en/web/authflowv2/view_recovery_code.html [hunk 3]
```diff
@@ -1,54 +1,48 @@
...
+		<code id="copy-button-source" class="hidden">{{ template "__recovery_code.html" . }}</code>
+		<div class="mt-5 grid gap-4 grid-cols-1 tablet:grid-cols-2">
+			<form class="hidden tablet:block" method="post" novalidate target="_blank">
+				{{ $.CSRFField }}
+				<button class="recovery-code__btn w-full" type="submit" name="x_action" value="download" data-turbo="false">
+				{{ template "download-button-label" }}
+				</button>
+			</form>
+			<button
+				class="recovery-code__btn w-full"
```
Comment (tung2744) [inline resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:30]: The button type here should be `tertiary-button`.

File: authui/src/authflowv2/components/recovery-code-list.css (hunk 1 above)
Comment (tung2744) [reply authui/src/authflowv2/components/recovery-code-list.css:18]: As mentioned in https://github.com/authgear/authgear-server/pull/3663/files#r1462773268 I think recovery-code__btn can be removed

File: resources/authgear/templates/en/web/authflowv2/view_recovery_code.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        <button
+          form="download-form"
+          class="tertiary-btn w-full"
+          type="submit"
+          name="x_action"
+          value="download"
+          data-turbo="false"
+        >
+          {{ template "download-button-label" }}
+        </button>
```
Comment (tung2744) [inline resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:31]: If I remember it correctly, in previous discussion we said we will only display download button in desktop.

@fungc-io Please correct me if I am wrong.

Comment (fungc-io) [reply resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:31]: yes, you are correct

Comment (tung2744) [reply resources/authgear/templates/en/web/authflowv2/view_recovery_code.html:32]: @IniZio I guess we should use `IsSupportedMobilePlatform` here

**Pull request merged**